Sources/Build/BuildPlan.swift

```diff
@@ -717,8 +728,13 @@ public final class SwiftTargetBuildDescription {
             // can construct the linker flags. In the future we will use a generated
             // code stub for the cases in which the linker doesn't support it, so that
             // we can rename the symbol unconditionally.
-            if buildParameters.linkerFlagsForRenamingMainFunction(of: target) != nil {
-                args += ["-Xfrontend", "-entry-point-function-name", "-Xfrontend", "\(target.c99name)_main"]
+            // No `-` for these flags because the set of Strings in driver.supportedFrontendFlags do
+            // not have a leading `-`
+            let flags: Set = ["entry-point-function-name"]
+            if SwiftTargetBuildDescription.checkSupportedFrontendFlags(flags: flags, fs: self.fs) {
```

Comment:
Nit: since it's just a set of one member, this could be further simplified to `flags: ["entry-point-function-name"]`.